src/BaseChallengeOptions.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AltchaOrg\Altcha;
+
+/**
+ * @phpstan-type ChallengeParams array<string, null|scalar>
+ */
+class BaseChallengeOptions
```

Comment:
I'm using a base class here because in PHP < 8 it's impossible to skip default parameters if you want specify another one.

That's the reason I implemented two child classes with different parameters (ordering) to allow a set of defined values, omitting those that should never be specified by users anyway.

In PHP > 8 I could have used a union type instead (no common base class for `ChallengeOptions` and `CheckChallengeOptions`) but the parameters are basically the same anyway and I didn't want to completely rewrite how challenge validation works.